hs_core/models.py

```diff
@@ -238,6 +237,66 @@ def page_permissions_page_processor(request, page):
         for edit_grp in edit_groups:
             edit_grp.can_undo = False
 
+    users_json = []
+
+    # TODO: simplify these iterations
+    for usr in owners:
+        users_json.append({
+            "user_type": "user",
+            "access": "owner",
+            "id": usr.id,
+            "pictureUrl": usr.userprofile.picture.url if usr.userprofile.picture else None,
+            "best_name": best_name(usr),
+            "user_name": usr.username,
+            "can_undo": usr.can_undo
+        })
+
+    for usr in editors:
+        users_json.append({
+            "user_type": "user",
+            "access": "edit",
+            "id": usr.id,
+            "pictureUrl": usr.userprofile.picture.url if usr.userprofile.picture else None,
+            "best_name": best_name(usr),
+            "user_name": usr.username,
+            "can_undo": usr.can_undo
+        })
+
+    for usr in viewers:
+        users_json.append({
+            "user_type": "user",
+            "access": "view",
+            "id": usr.id,
+            "pictureUrl": usr.userprofile.picture.url if usr.userprofile.picture else None,
+            "best_name": best_name(usr),
+            "user_name": usr.username,
+            "can_undo": usr.can_undo
+        })
+
+    for usr in edit_groups:
+        users_json.append({
+            "user_type": "group",
+            "access": "edit",
+            "id": usr.id,
+            "pictureUrl": usr.gaccess.picture.url if usr.gaccess.picture else None,
+            "best_name": usr.name,
+            "user_name": None,
+            "can_undo": usr.can_undo
+        })
+
+    for usr in view_groups:
+        users_json.append({
+            "user_type": "group",
+            "access": "view",
+            "id": usr.id,
+            "pictureUrl": usr.gaccess.picture.url if usr.gaccess.picture else None,
+            "best_name": usr.name,
+            "user_name": None,
+            "can_undo": usr.can_undo
+        })
+
+    users_json = json.dumps(users_json)
+
```

Comment:
seems like the Python `usr` User object should have an `access` field... that would be another PR likely, though to refactor all that and change necessary Models if that applies

Reply:
then, there wouldn't be separate Lists for Users by access type... and mapping from the Lists to the dict for JSON serialization would be a dictionary comprehension